src/Microsoft.AspNetCore.Razor.VSCode/src/TelemetryReporter.ts

```diff
@@ -39,6 +41,45 @@ export class TelemetryReporter {
         this.reportError('VSCode.Razor.ErrorOnActivation', error);
     }
 
+    public reportProjectInfo(project: IRazorProject) {
+        const projectConfiguration = project.configuration;
+        if (!projectConfiguration) {
+            // A project.razor.json file hasn't been created for the project yet.
+            return;
+        }
+
+        let configurationName: string;
+        let languageVersion: string;
+        if (projectConfiguration.configuration) {
+            configurationName = projectConfiguration.configuration.ConfigurationName;
+            languageVersion = projectConfiguration.configuration.LanguageVersion;
+        } else {
+            configurationName = 'Default';
+            languageVersion = 'Default';
+        }
```

Comment:
Don't know if you're into TypeScript tricks, but if you are, lines 51-59 could be replace with:

```typescript
const { ConfigurationName, LanguageVersion }
    = projectConfiguration.configuration || { ConfigurationName: 'Default', LanguageVersion: 'Default' };
```

Reply:
Though it is a bit weird that the incoming data is of the shape `project.configuration.configuration.someprops` (with `configuration` repeated). Not saying it's wrong - just a bit quirky!

Reply:
Not a fan of TypeScript tricks 😄